conans/model/conan_file.py

```diff
@@ -285,3 +290,65 @@ def __repr__(self):
             return "%s/%s@PROJECT" % (self.name, self.version)
         else:
             return "PROJECT"
+
+
+class ConanFileEditable(object):
+    """ Wrapper over a ConanFile object to handle an editable package """
+
+    def __init__(self, conanfile):
+        assert isinstance(conanfile, ConanFile)
+        self._conanfile = conanfile
+        self._package_layout_file = None
+
+    def __getattr__(self, item):
+        if item == 'package_info':
+            def package_info():
+                self._conanfile.package_info()
+                cpp_info = self._conanfile.cpp_info
+
+                content = load(self._package_layout_file)
+                base_path = os.path.dirname(self._package_layout_file)
+                data = parse_package_layout_content(content=content, base_path=base_path,
+                                                    settings=self.settings, options=self.options)
+
+                 # Replace directories with those in 'data'
+                for key, items in data.items():
+                    setattr(cpp_info, key, items)
+
+            return package_info
+
+        return getattr(self._conanfile, item)
+
+    def __setattr__(self, key, value):
+        if key == '_conanfile' or key == '_package_layout_file':
+            self.__dict__[key] = value
+            return
+        setattr(self._conanfile, key, value)
+
+    def set_package_layout_file(self, filepath):
+        self._package_layout_file = filepath
+
+
+def parse_package_layout_content(content, base_path, settings=None, options=None):
+    """ Returns a dictionary containing information about paths for a CppInfo object: includes,
+    libraries, resources, binaries,... """
+    ret = {k: [] for k in ['includedirs', 'libdirs', 'resdirs', 'bindirs']}
+
+    def _work_on_value(value, base_path_, settings_, options_):
+        value = re.sub(r'\\\\+', r'\\', value)
+        value = value.replace('\\', '/')
+        isabs = ntpath.isabs(value) or posixpath.isabs(value)
```

Comment:
Never have done that. Why not `os.path.isabs`?

Reply:
This way I can run the same tests (all) in windows and linux; otherwise I need to duplicate them and enable/disable some of them for win/linux.